src/services/linux-files.service.ts

```diff
@@ -5,14 +5,17 @@ import { IListDirParams } from '../interfaces/list-dir-params.interface';
 import { Observable } from 'rxjs';
 import { StreamService } from './stream.service';
 import { map } from 'rxjs/operators';
+import { PLATFORMS } from '../constants/main.constants';
+
+const isOSMacOS = () => process.platform === PLATFORMS.MAC_OS;
 
 export class LinuxFilesService extends FileService {
   constructor(private streamService: StreamService) {
     super();
   }
 
   getFolderSize(path: string): Observable<{}> {
-    const du = spawn('du', ['-s', '-b', path]);
+    const du = isOSMacOS() ? spawn('du', ['-s', path]) : spawn('du', ['-s', '-b', path]);
```

Comment:
Note the default is to return _blocks_. So this won’t give the right number on macOS (the reason for the original change to `-b`). AFAIK the default block size is 512b.

My suggestion is to use `-k` (kilobytes), and skip the bytes to kB call for macOS.

Reply:
Thanks for your comment @marcins! You mean use `-k` instead of `-s` on macOS?

Reply:
Oh wait, use `-sk` I guess, since we're just interested in the summary

Reply:
@marcins, you were absolutely right, it did not give the right number on macOS.

Actually, the solution was much easier than this: Simply use `du -sk` on _both_ Linux and macOS. :)